core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -484,18 +523,22 @@ public int hashCode()
             return Objects.hash(resolvedFunction, arguments, distinct, filter, orderingScheme, mask);
         }
 
-        private void verifyArguments(Step step)
+        private void verifyArguments(Step step, boolean isGlobalAggregation)
         {
             int expectedArgumentCount;
             if (step == SINGLE || step == Step.PARTIAL) {
                 expectedArgumentCount = resolvedFunction.getSignature().getArgumentTypes().size();
             }
-            else {
-                // Intermediate and final steps get the intermediate value and the lambda functions
+            else if (isGlobalAggregation) {
+                // Global intermediate and final steps get the intermediate value and the lambda functions
                 expectedArgumentCount = 1 + (int) resolvedFunction.getSignature().getArgumentTypes().stream()
                         .filter(FunctionType.class::isInstance)
                         .count();
             }
+            else {
+                // Hash intermediate and final steps get the intermediate value and all the arguments
```

Comment:
Do you have tests for lambda and adaptive aggregation

Reply:
General tests for lambda (e.g. `io.trino.testing.AbstractTestEngineOnlyQueries#testReduceAgg`) should cover case where PA is not disabled.
I added `io.trino.tests.TestAdaptivePartialAggregation#testAggregationWithLambda` to cover test wher PA is disabled